src/main/java/org/jboss/logmanager/handlers/AsyncHandler.java

```diff
@@ -146,6 +146,10 @@ protected void doPublish(final ExtLogRecord record) {
             // Copy the MDC over
             record.copyMdc();
         }
+        if (thread.getId() == record.getThreadID() && thread.getName().equals(record.getThreadName())) {
```

Comment:
One problem here is that we don't recognize `getLongThreadID` (which is `@since 16`), so given a long-enough running process (especially one using virtual threads), this check will eventually start failing and deadlocks will start up.

If we can't start using 17 as a baseline, then I think we need to figure out a way to support `getLongThreadID` in a compatible way across MR JARs.

Reply:
That's a good point. I suppose we could use the `JdkSpecific` and have some kind of: `JdkSpecific.isFromWorkerThread(Thread, LogRecord)`.

Reply:
Another edge case is the case of a deserialized log record... perhaps it would be a good idea to cache the `Thread` in a transient field?